src/DynamoCore/Core/Notifications.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+
+namespace Dynamo.Core
+{
+    public enum NotificationLevel { Mild, Moderate, Error }
+
+    /// <summary>
+    /// The Notifications class is used to raise notifications that
+    /// can be handled by the UI or the logger.
+    /// </summary>
+    public class Notifications
```

Comment:
The Notifications class is created to serve the specific purpose of raising notifications that can be handled anywhere in the system with a reference to DynamoCore.

Notifications should ultimately replace the use of the Logger for bubbling warnings up to the UI. The Logger should be exclusively for writing to the log. This is outside of the scope of this PR, but is covered in [this internal task](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6549).